polars/polars-time/src/windows/groupby.rs

```diff
@@ -400,7 +400,16 @@ pub(crate) fn groupby_values_iter_full_lookahead<'a>(
 
             let b = Bounds::new(lower, upper);
 
-            debug_assert!(i < time.len());
+            // find starting point of window
+            for &t in &time[i..] {
```

Comment:
Given that we have a nested loop now. 

Can we maybe keep the old behavior for `offset= 0`?

We can branch on the `offset=?` case and have a loop for both cases.

Or don't we hit this path on the `offset=0` case?

Reply:
good call, thanks - yes, but only if the window is closed on the left (if it's open on the left, then `time[i]` shouldn't be included in the group)